File: app/Models/Event.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Event extends Model
{
    use HasFactory;
    protected $fillable = [
        'title',
        'time_start',
        'time_end',
        'description',
        'location',
        'created_by'
    ];
    
    protected $appends = ['is_ongoing'];
    public function getIsOngoingAttribute()
    {
        $now = now();
        return $now->between($this->time_start, $this->time_end);
    }

    protected $casts = [
        'time_start' => 'datetime',
        'time_end' => 'datetime',
    ];
    public function creator() {
        return $this->belongsTo(User::class, 'created_by');
    }

    public function managers() {
        return $this->belongsToMany(User::class, 'event_managers');
    }

    public function participations()
    {
        return $this->hasMany(Participation::class);
    }
}

File: mobile_flutter/lib/models/event.dart
class Event {
  final int id;
  final String title;
  final String? description;
  final String timeStart;
  final String timeEnd;
  final String location;

  Event({
    required this.id,
    required this.title,
    this.description,
    required this.timeStart,
    required this.timeEnd,
    required this.location,
  });

  factory Event.fromJson(Map<String, dynamic> json) {
    return Event(
      id: json['id'],
      title: json['title'],
      description: json['description'],
      timeStart: json['time_start'],
      timeEnd: json['time_end'],
      location: json['location'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'title': title,
      'description': description,
      'time_start': timeStart,
      'time_end': timeEnd,
      'location': location,
    };
  }
}
